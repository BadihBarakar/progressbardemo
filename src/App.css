html,
body,
#root {
  height: 100%;
  margin: 0;
}

.App {
  height: 100%;
  width: 100%;
  /* text-align: center; */
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.App-link {
  color: #61dafb;
}
