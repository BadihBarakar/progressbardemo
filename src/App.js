import React from "react";
import "./App.css";
import ProgressBar from "./components/progressbar/progressbar";
import Controls from "./components/controles/controls";

import "./css/styles.css";

class App extends React.Component {
  state = {
    endPointParams: {},
    selectedBar: 0,
  };

  componentDidMount() {
    fetch("http://pb-api.herokuapp.com/bars")
      .then((result) => {
        return result.json(result.body);
      })
      .then((body) => {
        let { endPointParams } = this.state;
        endPointParams = body;
        this.setState({ endPointParams });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  handleButton = (btnIndex) => {
    let { endPointParams, selectedBar } = this.state;
    let barVal = endPointParams.bars[selectedBar];
    let buttonVal = endPointParams.buttons[btnIndex];
    barVal += buttonVal;
    barVal = barVal < 0 ? 0 : barVal;
    endPointParams.bars[selectedBar] = barVal;

    this.setState({ endPointParams });
  };

  handleSelectBar = (event) => {
    let selectedBar = this.state.selectedBar;
    selectedBar = event.currentTarget.value;

    this.setState({ selectedBar });
  };

  render() {
    const { endPointParams } = this.state;
    let containerHight = endPointParams.bars
      ? endPointParams.bars.length * 30 + 100 + 75
      : 100;
    return (
      <div className="App">
        <div className="container" style={{ height: containerHight }}>
          <div style={{ height: "3rem" }}>
            <h2>Progress Bar Demo</h2>
          </div>
          {endPointParams.bars
            ? endPointParams.bars.map((v, i) => {
                return (
                  <ProgressBar value={v} limit={endPointParams.limit} key={i} />
                );
              })
            : null}
          <Controls
            buttons={endPointParams.buttons ? endPointParams.buttons : null}
            bars={endPointParams.bars ? endPointParams.bars.length : null}
            handleButtonClick={this.handleButton}
            onSelectChange={this.handleSelectBar}
          />
        </div>
      </div>
    );
  }
}

export default App;
