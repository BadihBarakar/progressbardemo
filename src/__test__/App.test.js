import React from "react";
import ReactDOM from "react-dom";
import App from "../app";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

// it("render controls successfully", () => {
//   // test that the component will render successfully in another container...
//   const endPointParams = {
//     buttons: [14, 20, -35, -11],
//     bars: [57, 66],
//     limit: 190,
//   };

//   const div = document.createElement("div");
//   ReactDOM.render(
//     <Controls
//       buttons={endPointParams.buttons ? endPointParams.buttons : null}
//       bars={endPointParams.bars ? endPointParams.bars.length : null}
//       handleButtonClick={handleButton}
//       onSelectChange={handleSelectBar}
//     />,
//     div
//   );
//   ReactDOM.unmountComponentAtNode(div);
// });

// it("renders controls correctly", () => {
//   // test that the component contains all correct elements required...
//   const endPointParams = {
//     buttons: [5, 46, -32, -22],
//     bars: [40, 22, 87, 50],
//     limit: 160,
//   };

//   const component = render(
//     <Controls
//       buttons={endPointParams.buttons ? endPointParams.buttons : null}
//       bars={endPointParams.bars ? endPointParams.bars.length : null}
//       handleButtonClick={handleButton}
//       onSelectChange={handleSelectBar}
//     />
//   );
//   expect(component.getByTestId("mainDiv"));
//   expect(component.getByTestId("selectDiv"));
//   expect(component.getByTestId("selectProgBar"));
//   expect(component.getByTestId("selectProgBar"));
//   for (let n = 0; n < endPointParams.buttons.length; n++) {
//     expect(component.getByTestId("buttonDiv" + n));
//     expect(component.getByTestId("button" + n));
//   }
// });

it("match snapshot", () => {
  // test that the component matches the snapshot,
  // creating a snpshot if it doesn't exist...
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});
