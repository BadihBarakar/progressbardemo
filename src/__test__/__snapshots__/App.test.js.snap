// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
    style={
      Object {
        "height": 100,
      }
    }
  >
    <div>
      <h2>
        Progress Bar Demo
      </h2>
    </div>
  </div>
</div>
`;
